scripts/build-windows-provisioning-scripts.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash -e
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT license.
+
+required_env_vars=(
+    "CLIENT_ID"
+    "CLIENT_SECRET"
+    "RESOURCE_GROUP_NAME"
+    "STORAGE_ACCOUNT_NAME"
+    "STORAGE_CONTAINER_NAME"
+    "SUBSCRIPTION_ID"
+    "TENANT_ID"
+)
+
+for v in "${required_env_vars[@]}"
+do
+    if [ -z "${!v}" ]; then
+        echo "$v was not set!"
+        exit 1
+    fi
+done
+
+CREATE_TIME="$(date +%s)"
+az login --service-principal -u $CLIENT_ID -p $CLIENT_SECRET --tenant $TENANT_ID
+
+avail=$(az storage account check-name -n ${STORAGE_ACCOUNT_NAME} -o json | jq -r .nameAvailable)
```

Comment:
nit: for scripts like this I tend to lean towards using az built in querying method so folks don't have to have jq installed:

```
az storage account check-name -n ${STORAGE_ACCOUNT_NAME}  --query 'nameAvailable' -o tsv
```

Reply:
Most of the existing script content in this repo assumes jq is installed - this seems like an easy update tho so i'll try it out.